buildrunner/docker/multiplatform_image_builder.py

```diff
@@ -400,57 +298,33 @@ def _build_single_image(
         else:
             docker.buildx.build(
                 path,
-                tags=tagged_names,
+                tags=[image_ref],
                 platforms=[platform],
                 load=True,
-                file=file,
+                file=dockerfile,
                 build_args=build_args,
                 builder=builder,
                 cache=cache,
                 pull=pull,
                 **self._get_build_cache_options(builder),
             )
         # Push after the initial load to support remote builders that cannot access the local registry
-        docker.push(tagged_names)
+        docker.push([image_ref])
 
-        # Check that the images were built and in the registry
-        # Docker search is not currently implemented in python-on-wheels
-        image_id = None
-        for tag_name in tagged_names:
-            try:
-                images = docker.image.pull(tag_name)
-                assert images, f"Failed to build {tag_name}"
-                image_id = docker.image.inspect(tag_name).id
-                # Removes the image from host, if this fails it is considered a warning
-                try:
-                    LOGGER.debug(f"Removing {tag_name}")
-                    docker.image.remove(tag_name, force=True)
-                except python_on_whales.exceptions.DockerException as err:
-                    LOGGER.warning(f"Failed to remove {images}: {err}")
-            except python_on_whales.exceptions.DockerException as err:
-                LOGGER.error(f"Failed to build {tag_name}: {err}")
-                raise err
-
-        self._intermediate_built_images[mp_image_name].append(
-            ImageInfo(
-                repo=name,
-                tags=tags,
-                platform=platform,
-                digest=image_id,
-            )
-        )
+        # Retrieve the digest and put it on the queue
+        image_digest = self._get_image_digest(image_ref)
+        queue.put((image_ref, image_digest))
 
-    def get_single_platform_to_build(self, platforms: List[str]) -> str:
+    def _get_single_platform_to_build(self, platforms: List[str]) -> str:
         """Returns the platform to build for single platform flag"""
 
         assert isinstance(platforms, list), f"Expected list, but got {type(platforms)}"
         assert (
             len(platforms) > 0
         ), f"Expected at least one platform, but got {len(platforms)}"
 
-        host_system = system()
-        host_machine = machine()
-        native_platform = None
+        host_system = python_platform.system()
```

Comment:
Does this need to have `lower()` added? To match where this is done in the other code `native_system = python_platform.system().lower()`

Reply:
This is actually done in #110, which is a follow-up PR to this one. I had already written that portion of the PR at this point and so I was just getting something in there more as a placeholder. I can change it, but it'll be overwritten by the other PR anyways and cause some merge conflicts, so I'd like to avoid that if possible.

Reply:
After looking a little deeper, I realized I was looking at the wrong code. Yes, this could use some improvement. I've changed the comparison just below to use `lower()`, but kept this as the original value as it is potentially used in a platform comparison below.